pallets/amm/src/lib.rs

```diff
@@ -156,6 +159,38 @@ pub mod pallet {
                 let pool_liquidity_amount: PoolLiquidityAmount =
                     Self::pools(base_asset, quote_asset);
 
+                let optimal_quote_amount = Self::quote(
+                    base_amount,
+                    pool_liquidity_amount.base_amount,
+                    pool_liquidity_amount.quote_amount,
+                );
+
+                let (ideal_base_amount, ideal_quote_amount): (Balance, Balance) =
+                    if optimal_quote_amount <= quote_amount {
+                        (base_amount, optimal_quote_amount)
+                    } else {
+                        let optimal_base_amount = Self::quote(
+                            quote_amount,
+                            pool_liquidity_amount.base_amount,
+                            pool_liquidity_amount.quote_amount,
+                        );
```

Comment:
I think the base and quote amounts need to be inverse from the `optimal_quote_amount` in order to calculate `optimal_base_amount`


```rust
let optimal_base_amount = Self::quote(
    quote_amount,
    pool_liquidity_amount.quote_amount,
    pool_liquidity_amount.base_amount,
);
```

[similar to uniswap](https://github.com/Uniswap/uniswap-v2-periphery/blob/dda62473e2da448bc9cb8f4514dadda4aeede5f4/contracts/UniswapV2Router02.sol#L54)

Reply:
I second that, this seems correct